docs/ir/ir-breaking-changes.rst

```diff
@@ -4,3 +4,32 @@ Solidity IR-based Codegen Changes
 
 This section highlights the main differences between the old and the IR-based codegen,
 along with the reasoning behind the changes and how to update affected code.
+
+Semantic Only Changes
+=====================
+
+This section lists the changes that are semantic-only, thus potentially
+hiding new and different behavior in existing code.
+
+ * When storage structs are deleted, every storage slot that contains a member of the struct is set to zero entirely. Formally, padding space was left untouched.
+Consequently, if the padding space within a struct is used to store data (e.g. in the context of a contract upgrade), you have to be aware that ``delete`` will now also clear the added member (while it wouldn't have been cleared in the past).
```

Comment:
Maybe we should rather say "that old code generated for ``delete`` before adding that struct" or something like that, but I'd also say it's fine for now - we'll need to refine this in the end when we make it "public" anyways I expect.

Reply:
Ah, you merged already anyways :-D.